edit-post/components/layout/index.js

```diff
@@ -52,6 +53,7 @@ function Layout( {
 		<div className={ className }>
 			<DocumentTitle />
 			<UnsavedChangesWarning />
+			<MetaBoxesUnsavedWarning />
```

Comment:
The logic to prevent from leaving the page needed a separate component for preventing leaving the page when the meta boxes are dirty.